Mu2eKinKal/inc/KKStrawXing.hh

```diff
@@ -87,9 +108,37 @@ namespace mu2e {
         sxconfig_.hitstate_ = shptr_->hitState();
       else
         sxconfig_.hitstate_ = WireHitState::inactive;
+      if(sxconfig_.scalevar_)
+        varscale_ = miconfig.varianceScale();
     }
     // find the material xings from gas, straw wall, and wire
-    smat_.findXings(tpca_.tpData(),sxconfig_,EXING::matXings());
+    smat_.findXings(tpca_.tpData(),sxconfig_,mxings_);
+    // reset
+    fparams_ = Parameters();
+    if(mxings_.size() > 0){
+      // compute the parameter effect for forwards time
+      std::array<double,3> dmom = {0.0,0.0,0.0}, momvar = {0.0,0.0,0.0};
+      this->materialEffects(TimeDir::forwards, dmom, momvar);
```

Comment:
Is "this" necessary?  I see other places, not introduced in this PR, where it also might not be needed.

Reply:
'this' is need to access base class members of templated classes, even if the functions aren't overridden.  IMHO It is a flaw in C++, though I've read plenty of online commentary that 'explains' it.  The alternative is explicitly call the function, ie BaseClass::materialEffects(.  There is no consensus in the community on best practices, I chose to use 'this'.